### DISCUSSION TITLE
Hydrogen bond calculation

### DISCUSSION THREAD

**Discussion by horlust** [MDAnalysis/mdanalysis#4613] [Q&A] [answered]
Hi everyone!

I am computing the hydrogen bonds within a system containing 510 water molecules. I do something like this:

```
hbonds = HBA(universe=u, d_a_cutoff=3.5, d_h_a_angle_cutoff=30)

hydrogens_sel = hbonds.guess_hydrogens("resname SOL")
acceptors_sel = hbonds.guess_acceptors("resname SOL")
hydrogen_donors_sel = "resname SOL and name OW"

hbonds.hydrogens_sel = f"({hydrogens_sel})"
hbonds.acceptors_sel = f"({acceptors_sel})"
hbonds.donors_sel = f"({hydrogen_donors_sel})"

hbonds.run(verbose=True)
```

However, the results are:

![image](https://github.com/MDAnalysis/mdanalysis/assets/50314998/ddf723c6-beb3-4618-889f-a650af412c98)

The amount of hydrogen bonds per molecule of water is of around 9. It is very unlikely, and different from the output I get using GROMACS:

![image](https://github.com/MDAnalysis/mdanalysis/assets/50314998/a4e96900-4f22-4698-b461-1416f5617427)

Do you know what could be the problem?

Thanks!

**Comment by remaker789** [comment, marked as answer]
The definitions of angle are different between MDAnalysis and gmx hbond. MDAnalysis uses the D-H-A angle(see [this](https://userguide.mdanalysis.org/stable/examples/analysis/hydrogen_bonds/hbonds.html)), but gmx hbond use the H-D-A angle(see [this](https://manual.gromacs.org/current/reference-manual/analysis/hydrogen-bonds.html)). You can use 'd_h_a_angle_cutoff=150' in MDAnalysis, however, the results of hydrogen bonds from MDAnalysis will still be less than gmx bond due to the definition of angle.
